src/main/java/jenkins/plugins/git/GitSCMSourceContext.java

```diff
@@ -272,6 +317,9 @@ public final C withRemoteName(String remoteName) {
     @NonNull
     public final List<RefSpec> asRefSpecs() {
         List<RefSpec> result = new ArrayList<>(Math.max(refSpecs.size(), 1));
+        if (wantOtherRefs() && wantBranches()) {
```

Comment:
what happens if only one of these is true?

Reply:
Well if it want other refs, then that refspec is already added to the list retrieved below. Which means that if you also want branches that refspec needs to be added, otherwise you will only fetch the other refs and not the branches.

So if we don't want other refs (and no other explicit refspec is set) then the list of refspecs is empty and so the branches will be fetched as by default.

Reply:
I.e. this is an ugly hack to get it to work when you both want other refs and branches, if any of those are false we can go on with our normal flow. :)

Reply:
thanks for the clarifying comment